pkg/southbound/synchronizer/factory.go

```diff
@@ -30,7 +31,8 @@ import (
 // These synchronizers then listen out for configEvents relative to a device and
 func Factory(changeStore *store.ChangeStore, configStore *store.ConfigurationStore, deviceStore *topocache.DeviceStore,
 	topoChannel <-chan events.TopoEvent, opStateChan chan<- events.OperationalStateEvent,
-	errChan chan<- events.DeviceResponse, dispatcher *dispatcher.Dispatcher, modelReadOnlyPaths modelregistry.GlobalReadOnlyPaths) {
+	errChan chan<- events.DeviceResponse, dispatcher *dispatcher.Dispatcher,
+	registry *modelregistry.ModelRegistry, operationalStateCache map[topocache.ID]map[string]*change.TypedValue) {
```

Comment:
do you need to send down the whole model registry?

Reply:
refactored